messaginginapp/src/test/java/io/customer/messaginginapp/ModuleMessagingInAppTest.kt

```diff
@@ -184,4 +187,54 @@ internal class ModuleMessagingInAppTest : JUnitTest() {
         val gistProperties = GistMessageProperties.getGistProperties(capturedMessage)
         assert(gistProperties.position == MessagePosition.CENTER)
     }
+
+    @Test
+    fun onAction_givenNonCloseAction_expectTrackInAppMetricEventPublished() {
+        val message = createInAppMessage(campaignId = "test_campaign_id")
+        val inAppMessage = InAppMessage(
+            messageId = message.messageId,
+            deliveryId = "test_campaign_id",
+            queueId = message.queueId
+        )
+
+        mockkObject(InAppMessage.Companion)
+        every { InAppMessage.getFromGistMessage(any()) } returns inAppMessage
+
+        module.initialize()
+        module.onAction(message, "current_route", "test_action", "Test Action")
+
+        verify(exactly = 1) {
+            inAppEventListenerMock.messageActionTaken(inAppMessage, "test_action", "Test Action")
+            eventBus.publish(
+                Event.TrackInAppMetricEvent(
+                    deliveryID = "test_campaign_id",
+                    event = Metric.Clicked,
+                    params = mapOf("action_name" to "Test Action", "action_value" to "test_action")
+                )
+            )
+        }
+    }
+
+    @Test
+    fun onAction_givenCloseAction_expectNoTrackInAppMetricEventPublished() = runTest {
```

Comment:
Can we add `ExperimentalCoroutinesApi` annotation to fix the warning?

```
This declaration needs opt-in. Its usage should be marked with '@kotlinx.coroutines.ExperimentalCoroutinesApi' or '@OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)'
```

Reply:
can do